File: engine/src/core/artifacts/ArtifactManifest.ts
import type { ArtifactMeta } from "./ArtifactTypes.js";

export type ManifestSchemaVersion = "1.0";

export interface ManifestInputFile {
	path: string;
	sha256: string;
	size: number;
	kind?: string;
}

export interface ArtifactManifest {
	schemaVersion: ManifestSchemaVersion;
	runId: string;
	projectRoot: string;
	adapter: string;

	createdAt: string; // ISO string

	inputs: {
		files: ManifestInputFile[];
	};

	// outputs[artifactId] = metadata
	outputs: Record<string, ArtifactMeta>;
}

File: ui/lib/models/artifact_manifest.dart
class ArtifactManifest {
  final String schemaVersion;
  final String runId;
  final String projectRoot;
  final String adapter;
  final String createdAt;
  final Map<String, dynamic> inputs;
  final Map<String, dynamic> outputs;

  ArtifactManifest({
    required this.schemaVersion,
    required this.runId,
    required this.projectRoot,
    required this.adapter,
    required this.createdAt,
    required this.inputs,
    required this.outputs,
  });

  factory ArtifactManifest.fromJson(Map<String, dynamic> json) {
    return ArtifactManifest(
      schemaVersion: json['schemaVersion'] as String,
      runId: json['runId'] as String,
      projectRoot: json['projectRoot'] as String,
      adapter: json['adapter'] as String,
      createdAt: json['createdAt'] as String,
      inputs: json['inputs'] as Map<String, dynamic>,
      outputs: json['outputs'] as Map<String, dynamic>,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'schemaVersion': schemaVersion,
      'runId': runId,
      'projectRoot': projectRoot,
      'adapter': adapter,
      'createdAt': createdAt,
      'inputs': inputs,
      'outputs': outputs,
    };
  }
}
